packages/nimbus/src/components/text-input/text-input.docs.spec.tsx

```diff
@@ -0,0 +1,223 @@
+import { describe, it, expect, vi } from "vitest";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { TextInput, NimbusProvider } from "@commercetools/nimbus";
+
+/**
+ * @docs-section basic-rendering
+ * @docs-title Basic Rendering Tests
+ * @docs-description Verify the component renders with expected elements
+ * @docs-order 1
+ */
```

Comment:
JSDOC comments here allow us to formulate the transformed tests into  documentation in a structured way.